smart-metastore/src/main/java/org/smartdata/metastore/utils/MetaStoreUtils.java

```diff
@@ -353,6 +353,23 @@ public static MetaStore getDBAdapter(
           LOG.warn("Database URL not specified, using " + purl);
         }
 
+        if (purl.startsWith(MetaStoreUtils.MYSQL_URL_PREFIX)) {
+          String dbName;
+          String[] dbNameNotAllowed = new String[]{
+                  "mysql", "sys", "information_schema", "INFORMATION_SCHEMA", "performance_schema", "PERFORMANCE_SCHEMA"
+          };
+          if (purl.contains("?")) {
+            dbName = purl.substring(purl.indexOf("/", 13) + 1, purl.indexOf("?"));
```

Comment:
I think this line of code is very risky. Is there any other proper way to do the extraction? Or any other util tool can recognize the pattern?

Reply:
A proper method is used to get the db name in the updated code. Thanks!